tedpermission/src/main/java/com/gun0912/tedpermission/TedPermissionBase.java

```diff
@@ -40,8 +41,8 @@ private static boolean isGranted(Context context, @NonNull String permission) {
         return ContextCompat.checkSelfPermission(context, permission) == PackageManager.PERMISSION_GRANTED;
     }
 
-    public static ArrayList<String> getDeniedPermissions(Context context, @NonNull String... permissions) {
-        ArrayList<String> deniedPermissions = new ArrayList<>();
+    public static List<String> getDeniedPermissions(Context context, @NonNull String... permissions) {
+        List<String> deniedPermissions = new ArrayList<>();
```

Comment:
Is there a reason to change this?
It is a local variable and there is no connection point with the outside.
And, it is instance of `ArrayList`

Reply:
It was just base on `program to interfaces, not implementations`.

Although this request changed the listener used by the` ArrayList` to `List`, it was consistent under the same principle.